File: 182_weekly/2将二进制表示减到 1 的步骤数.cpp
class Solution {
public:
    int numSteps(string s) {
        int n = s.length();
        if(n == 1) return 0;
        int k = n - 1;
        int cnt = 0;
        while(k > 0){
            if(s[k] == '0') cnt++, k--;
            else{
                while(k >= 0 && s[k] == '1' ) cnt++, k--;
                cnt++;
                if(k >= 0) s[k] = '1';
            }
        }
        return cnt;
    }
};

File: 182_weekly/2将二进制表示减到 1 的步骤数.py
python 自带高精度
class Solution:
    def numSteps(self, s: str) -> int:
        num = 0;
        for d in s:
            num = num * 2 + ord(d) - ord('0')
        cnt = 0
        while num > 1:
            if num % 2 == 0:
                num //= 2
            else:
                num += 1
            cnt += 1
        return cnt